engine/src/test/scala/cromwell/SimpleWorkflowActorSpec.scala

```diff
@@ -1,80 +1,144 @@
 package cromwell
 
+import java.util.UUID
+
+import akka.actor.{Actor, ActorRef, Props}
 import akka.testkit._
-import cromwell.core.WorkflowState
+import com.typesafe.config.ConfigFactory
+import cromwell.SimpleWorkflowActorSpec._
+import cromwell.core.WorkflowId
 import cromwell.engine._
-import cromwell.engine.workflow.WorkflowDescriptorBuilder
+import cromwell.engine.workflow.WorkflowActor
+import cromwell.engine.workflow.WorkflowActor.{StartNewWorkflow, StartWorkflowCommand}
+import cromwell.services.MetadataEvent
+import cromwell.services.MetadataServiceActor.PutMetadataAction
 import cromwell.util.SampleWdl
 import cromwell.util.SampleWdl.HelloWorld.Addressee
 
 import scala.concurrent.duration._
+import scala.concurrent.{Await, Promise}
 import scala.language.postfixOps
 
 
-class SimpleWorkflowActorSpec extends CromwellTestkitSpec with WorkflowDescriptorBuilder {
+object SimpleWorkflowActorSpec {
 
-  override implicit val actorSystem = system
+  object MetadataWatchActor {
+    def props(matcher: Option[FailureMatcher], promise: Promise[Unit]): Props = Props(MetadataWatchActor(matcher, promise))
+  }
 
-//  private def buildWorkflowFSMRef(sampleWdl: SampleWdl, rawInputsOverride: String):
-//  TestFSMRef[WorkflowState, WorkflowData, OldStyleWorkflowActor] = {
-//    val workflowSources = WorkflowSourceFiles(sampleWdl.wdlSource(), rawInputsOverride, "{}")
-//    val descriptor = materializeWorkflowDescriptorFromSources(workflowSources = workflowSources)
-//    TestFSMRef(new OldStyleWorkflowActor(descriptor))
-//  }
+  case class MetadataWatchActor(matcher: Option[FailureMatcher], promise: Promise[Unit]) extends Actor {
+    override def receive = matcher match {
+      case None =>
+        promise.trySuccess(())
+        Actor.emptyBehavior
+      case Some(m) => {
+        case PutMetadataAction(events) if m.matches(events) =>
```

Comment:
Ah okay, I should add comments then.  This is making sure that the right metadata messages have been emitted by the WorkflowActor.  Normally there's a WorkflowManagerActor involved which spins up a real service registry, but this is testing at a lower level and I was only interested in seeing that the right messages make it to the service registry.